GiniSwitchSDK/GiniSwitchSDK/Classes/Extraction.swift

```diff
@@ -6,52 +6,15 @@
 //
 //
 
-public class  Extraction {
+public class  Extraction<T: ExtractionValue> : Codable {
 
-    public let name:String
-    public var value:ExtractionValue = ExtractionValue(value: "", unit: nil)
-    public var alternatives:[ExtractionValue] = []
+    public var value:T? = nil
+    public var alternatives:[T] = []
     
     public var valueString:String {
-        return value.valueString
+        return value?.valueString ?? ""
```

Comment:
My idea was to have the `valueString` method as sort of a `toString` that can be used in UI classes to display the extraction. Therefore, I didn't want nils to be possible.

For any "real" work, the `value` property should be used.

But, I also might consider removing this method because I'm not sure how useful it is.